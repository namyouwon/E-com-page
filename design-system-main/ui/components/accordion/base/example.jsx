// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import uniqueId from 'lodash.uniqueid';
import Accordion, { AccordionSection } from '../';
import { ActionOverflow } from '../../../components/menus/action-overflow/example';
import Card from '../../cards/';

export const AccordionExample = () => (
  <Accordion>
    <AccordionSection
      summary="Accordion summary"
      isOpen
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - A
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - B
    </AccordionSection>
    <AccordionSection
      summary="Accordion summary"
      referenceId={uniqueId('referenceId-')}
    >
      Accordion details - C
    </AccordionSection>
  </Accordion>
);

export default <AccordionExample />;

export let states = [
  {
    id: 'section-one-open',
    label: 'Section one open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-two-open',
    label: 'Section two open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-three-open',
    label: 'Section three open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'multiple-sections-open',
    label: 'Multiple sections open',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];

export let examples = [
  {
    id: 'long-heading-truncated',
    label: 'Long Section Heading with Truncation for larger screens',
    demoStyles: 'max-width: 500px;',
    storybookStyles: true,
    element: (
      <Card>
        <Accordion>
          <AccordionSection
            summary="Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </Card>
    )
  },
  {
    id: 'nested-accordions',
    label: 'Nested Accordions',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          <Accordion>
            <AccordionSection
              summary="Nested accordion summary"
              isOpen
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - A
            </AccordionSection>
            <AccordionSection
              summary="Nested accordion summary"
              referenceId={uniqueId('referenceId-')}
            >
              Accordion details - B
            </AccordionSection>
          </Accordion>
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'styled',
    label: 'Wrapped in Card',
    element: (
      <div className="slds-card">
        <Accordion>
          <AccordionSection
            summary="Accordion summary"
            isOpen
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - A
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - B
          </AccordionSection>
          <AccordionSection
            summary="Accordion summary"
            referenceId={uniqueId('referenceId-')}
          >
            Accordion details - C
          </AccordionSection>
        </Accordion>
      </div>
    )
  },
  {
    id: 'has-action-overflow',
    label: 'Has Action Overflow Menu',
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          actionMenu={
            <ActionOverflow position="right" dropdownIsOpen={false} />
          }
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Background Color',
    demoStyles: '--slds-c-accordion-summary-color-background: lightblue',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  },
  {
    id: 'section-background-color',
    label: 'Accordion Section - Text Color',
    demoStyles:
      '--slds-c-accordion-heading-text-color: green; --slds-c-accordion-heading-text-color-hover: red',
    storybookStyles: true,
    element: (
      <Accordion>
        <AccordionSection
          summary="Accordion summary"
          isOpen
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - A
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - B
        </AccordionSection>
        <AccordionSection
          summary="Accordion summary"
          referenceId={uniqueId('referenceId-')}
        >
          Accordion details - C
        </AccordionSection>
      </Accordion>
    )
  }
];
