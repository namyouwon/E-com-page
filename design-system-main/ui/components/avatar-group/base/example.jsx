// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import {
  AvatarGroup,
  AvatarGroupContent,
  AvatarGrouped,
  AvatarImage,
  AvatarGroupedIcon,
  AvatarGroupedUserInitials
} from '../';

export const ContextInversed = props => (
  <div
    className="demo-only"
    style={{ padding: '0.5rem', background: '#16325c' }}
  >
    {props.children}
  </div>
);

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  }
];

export const examples = [
  {
    id: 'with-icons',
    label: 'With icons',
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'with-user-initials',
    label: 'With initials',
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'with-user-initials-inversed',
    label: 'With initials inversed',
    demoStyles: 'padding: 0.5rem; background: #16325c',
    storybookStyles: true,
    element: (
      <AvatarGroup>
        <AvatarGroupContent type="user-initials" isInverse />
      </AvatarGroup>
    )
  },
  {
    id: 'with-image-and-icon',
    label: 'With image and icon',
    element: (
      <AvatarGroup>
        <AvatarGrouped isPrimary>
          <AvatarImage />
        </AvatarGrouped>
        <AvatarGrouped>
          <AvatarGroupedIcon />
        </AvatarGrouped>
      </AvatarGroup>
    )
  },
  {
    id: 'with-icon-and-user-initials',
    label: 'With icon and user initials',
    element: (
      <AvatarGroup>
        <AvatarGrouped isPrimary>
          <AvatarGroupedIcon />
        </AvatarGrouped>
        <AvatarGrouped>
          <AvatarGroupedUserInitials />
        </AvatarGrouped>
      </AvatarGroup>
    )
  },
  {
    id: 'with-user-initials-and-image',
    label: 'With user initials and image',
    element: (
      <AvatarGroup>
        <AvatarGrouped isPrimary>
          <AvatarGroupedUserInitials />
        </AvatarGrouped>
        <AvatarGrouped>
          <AvatarImage />
        </AvatarGrouped>
      </AvatarGroup>
    )
  },
  {
    id: 'x-small-image',
    label: 'X-Small',
    element: (
      <AvatarGroup size="x-small">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'small-image',
    label: 'Small',
    element: (
      <AvatarGroup size="small">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'medium-image',
    label: 'Medium',
    element: (
      <AvatarGroup size="medium">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'large-image',
    label: 'Large',
    element: (
      <AvatarGroup size="large">
        <AvatarGroupContent type="image" />
      </AvatarGroup>
    )
  },
  {
    id: 'x-small-icon',
    label: 'X-Small With icon',
    element: (
      <AvatarGroup size="x-small">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'small-icon',
    label: 'Small With icon',
    element: (
      <AvatarGroup size="small">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'medium-icon',
    label: 'Medium With icon',
    element: (
      <AvatarGroup size="medium">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'large-icon',
    label: 'Large With icon',
    element: (
      <AvatarGroup size="large">
        <AvatarGroupContent type="icon" />
      </AvatarGroup>
    )
  },
  {
    id: 'x-small-user-initials',
    label: 'X-Small With initials',
    element: (
      <AvatarGroup size="x-small">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'small-user-initials',
    label: 'Small With initials',
    element: (
      <AvatarGroup size="small">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'medium-user-initials',
    label: 'Medium With initials',
    element: (
      <AvatarGroup size="medium">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  },
  {
    id: 'large-user-initials',
    label: 'Large With initials',
    element: (
      <AvatarGroup size="large">
        <AvatarGroupContent type="user-initials" />
      </AvatarGroup>
    )
  }
];
