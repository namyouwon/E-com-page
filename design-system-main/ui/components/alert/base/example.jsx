// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import Alert from '../';
import { UtilityIcon } from '../../icons/base/example';
import EmptyLink from '../../../shared/empty-link/';
import Heading from '../../../shared/heading/';

export default [
  {
    id: 'default',
    label: 'Default',
    element: (
      <Alert>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];

export let states = [
  {
    id: 'theme-info',
    label: 'Default (Theme Class  - Deprecated)',
    element: (
      <Alert isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="user"
        />
        <Heading level="2">
          Logged in as John Smith (johnsmith@acme.com).{' '}
          <EmptyLink>Log out</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'warning',
    label: 'Warning',
    element: (
      <Alert type="warning">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-warning',
    label: 'Warning (Theme Class  - Deprecated)',
    element: (
      <Alert type="warning" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="warning"
        />
        <Heading level="2">
          Your browser is outdated. Your Salesforce experience may be degraded.{' '}
          <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'error',
    label: 'Error',
    element: (
      <Alert type="error">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-error',
    label: 'Error (Theme Class  - Deprecated)',
    element: (
      <Alert type="error" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="error"
        />
        <Heading level="2">
          Your browser is currently not supported. Your Salesforce may be
          degraded. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'offline',
    label: 'Offline',
    element: (
      <Alert type="offline">
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  },
  {
    id: 'theme-offline',
    label: 'Offline (Theme Class  - Deprecated)',
    element: (
      <Alert type="offline" isLegacy>
        <UtilityIcon
          containerClassName="slds-m-right_x-small"
          className="slds-icon_x-small"
          assistiveText={false}
          symbol="offline"
        />
        <Heading level="2">
          You are in offline mode. <EmptyLink>More Information</EmptyLink>
        </Heading>
      </Alert>
    )
  }
];
