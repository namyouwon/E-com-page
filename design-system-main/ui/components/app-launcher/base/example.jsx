// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from '../../../shared/svg-icon';
import {
  Modal,
  ModalHeader,
  ModalContent,
  Backdrop
} from '../../modals/base/example';
import {
  Section,
  SectionContent,
  SectionTitle,
  SectionTitleAction
} from '../../expandable-section/base/example';
import { Avatar } from '../../avatar/base/example';
import { ButtonIconStateful } from '../../button-icons/stateful/example';
import { Tooltip } from '../../tooltips/base/example';
import classNames from 'classnames';

const exampleDemoStyles = 'height: 800px;';

export const Context = props => (
  <div style={{ height: '800px' }}>{props.children}</div>
);

/// ///////////////////////////////////////////
// Partial(s)
/// ///////////////////////////////////////////
export let AppLauncherModal = props => (
  <Modal
    className="slds-modal_large slds-app-launcher"
    aria-labelledby="header43"
  >
    <ModalHeader className="slds-grid slds-grid_align-spread slds-grid_vertical-align-center">
      <h2 id="header43" className="slds-text-heading_medium">
        App launcher
      </h2>
      <div className="slds-app-launcher__header-search">
        <div className="slds-form-element">
          <label
            htmlFor="app-launcher-search"
            className="slds-form-element__label slds-assistive-text"
          >
            Find an app
          </label>
          <div className="slds-form-element__control slds-input-has-icon slds-input-has-icon_left">
            <SvgIcon
              className="slds-input__icon"
              sprite="utility"
              symbol="search"
            />
            <input
              type="search"
              className="slds-input"
              id="app-launcher-search"
              placeholder="Find an app"
              defaultValue={props.searchTerm}
            />
          </div>
        </div>
      </div>
      <button className="slds-button slds-button_neutral">App Exchange</button>
    </ModalHeader>
    <ModalContent className="slds-app-launcher__content slds-p-around_medium">
      <Section className="slds-is-open">
        <SectionTitle>
          <SectionTitleAction isOpen referenceId="appsContent">
            All Apps
          </SectionTitleAction>
        </SectionTitle>
        <SectionContent isOpen referenceId="appsContent">
          <div
            className="slds-assistive-text"
            id="drag-live-region"
            aria-live="assertive"
          >
            {props.dragDropLiveRegion}
          </div>
          <ul className="slds-grid slds-grid_pull-padded slds-wrap">
            {props.appTiles.map((tile, i) => {
              return (
                <li
                  className="slds-p-horizontal_small slds-size_1-of-1 slds-medium-size_1-of-3"
                  key={i}
                >
                  <AppLauncherTile
                    figureClass={tile.figureClass}
                    grabbed={tile.grabbed}
                    objectInitials={tile.initials}
                    label={tile.label}
                    description={tile.description}
                    hasTooltip={tile.hasTooltip}
                    tooltipText={tile.tooltipText}
                    index={i}
                  />
                </li>
              );
            })}
          </ul>
        </SectionContent>
      </Section>
      <hr />
      <Section className="slds-is-open">
        <SectionTitle>
          <SectionTitleAction isOpen referenceId="itemsContent">
            All Items
          </SectionTitleAction>
        </SectionTitle>
        <SectionContent isOpen referenceId="itemsContent">
          <ul className="slds-grid slds-wrap">
            {props.itemTiles.map((tile, i) => {
              return (
                <li
                  className="slds-col_padded slds-p-vertical_xx-small slds-size_1-of-5"
                  key={i}
                >
                  <a
                    href="#"
                    className="slds-truncate"
                    title={tile.label}
                    onClick={e => e.preventDefault()}
                  >
                    {tile.label}
                  </a>
                </li>
              );
            })}
          </ul>
        </SectionContent>
      </Section>
    </ModalContent>
  </Modal>
);

export const AppLauncherTile = props => (
  <div
    draggable
    className={classNames(
      'slds-app-launcher__tile slds-text-link_reset slds-is-draggable',
      props.className,
      {
        'slds-is-grabbed': props.grabbed
      }
    )}
  >
    <div className="slds-app-launcher__tile-figure">
      <Avatar className="slds-avatar_large">
        <abbr
          className={classNames('slds-avatar__initials', props.figureClass)}
          title={props.label}
        >
          {props.objectInitials}
        </abbr>
      </Avatar>
      <div className="slds-m-top_xxx-small">
        <ButtonIconStateful
          aria-pressed={props.grabbed ? 'true' : 'false'}
          assistiveText="Reorder"
          selected={props.grabbed}
          symbol="rows"
          title="Reorder"
        />
      </div>
    </div>
    <div className="slds-app-launcher__tile-body">
      <a href="#" onClick={e => e.preventDefault()}>
        {props.label}
      </a>
      <p>
        {props.description}{' '}
        {props.hasTooltip && (
          <button
            aria-describedby={`help-${props.index}`}
            className="slds-button slds-button_reset slds-text-link"
          >
            More
          </button>
        )}
      </p>
      <Tooltip
        className={classNames('slds-nubbin_top-right', {
          'slds-hide': !props.tooltipText
        })}
        id={`help-${props.index}`}
        style={{ position: 'absolute', top: '80px', right: '30px' }}
      >
        {props.tooltipText}
      </Tooltip>
    </div>
  </div>
);

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

/*
 * DragDropId relates app launcher tile anchor aria-describedby to the div that holds the instructions for drag & drop
 */
export const itemTiles = [
  { label: 'Accounts', symbol: 'account' },
  { label: 'Announcements', symbol: 'announcement' },
  { label: 'Approvals', symbol: 'approval' },
  { label: 'Campaigns', symbol: 'campaign' },
  { label: 'Cases', symbol: 'case' },
  { label: 'Coaching', symbol: 'coaching' },
  { label: 'Contacts', symbol: 'contact' }
];

export const searchItemTiles = [
  {
    label: (
      <span>
        <mark>Sales</mark> Invoices
      </span>
    ),
    symbol: 'account'
  },
  {
    label: (
      <span>
        <mark>Sales</mark> Objects
      </span>
    ),
    symbol: 'announcement'
  }
];

/*
 * Tile data for All Apps section inside example.
 * Using data instead of hard-coding becase we want to move the position of one App Tile based on state (ie. moved, dropped).
 */
export const appTiles = [
  {
    description:
      'The primary internal Salesforce org. Used to run our online sales business...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-27',
    grabbed: false,
    initials: 'SC',
    label: 'Sales Cloud'
  },
  {
    description: 'Salesforce Marketing Cloud lets businesses of any size...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-59',
    grabbed: false,
    initials: 'MC',
    label: 'Marketing Cloud'
  },
  {
    description: 'Community for managing employee benefits and time off.',
    hasTooltip: false,
    figureClass: 'slds-icon-custom-10',
    grabbed: false,
    initials: 'HR',
    label: 'HR Concierge'
  },
  {
    description: 'Manage your finances across multiple financial platforms...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-6',
    grabbed: false,
    initials: 'MM',
    label: 'My Money'
  },
  {
    description:
      'The key to call center and contact center management is more...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-91',
    grabbed: false,
    initials: 'CC',
    label: 'Call Center'
  },
  {
    description:
      'Areas of Focus are used to track customer support for your...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-50',
    grabbed: false,
    initials: 'CS',
    label: 'Customer Support Community'
  }
];

export const appTilesSearch = [
  {
    description: 'The primary internal Salesforce org. Used to run our...',
    hasTooltip: true,
    figureClass: 'slds-icon-custom-27',
    grabbed: false,
    initials: 'SC',
    label: (
      <span>
        <mark>Sales</mark> Cloud
      </span>
    )
  }
];

/*
 * Helpers to position the App Tile for each state (ie. moved, dropped, etc.)
 */
const moveItemPosition = function(arr, from, to) {
  let copyArr = arr.slice(0);
  copyArr.splice(to, 0, copyArr.splice(from, 1)[0]);
  return copyArr;
};

export const appTilesGrabbed = appTiles.slice(0);
appTilesGrabbed[0] = Object.assign({}, appTilesGrabbed[0], { grabbed: true });
export const appTilesMoved = moveItemPosition(appTiles, 0, 2);
appTilesMoved[2] = Object.assign({}, appTilesMoved[2], { grabbed: true });
export const appTilesDropped = moveItemPosition(appTiles, 0, 3);

const defaultComponent = (
  <Backdrop>
    <AppLauncherModal
      appTiles={appTiles}
      dragDropLiveRegion=""
      itemTiles={itemTiles}
    />
  </Backdrop>
);

export default defaultComponent;

export let states = [
  {
    id: 'default',
    label: 'Default',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: defaultComponent
  },
  {
    id: 'grabbed',
    label: 'Tile grabbed',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <Backdrop>
        <AppLauncherModal
          appTiles={appTilesGrabbed}
          dragDropLiveRegion="Sales Cloud: current position 1 of 6. Use the up and down arrows to move this app"
          grabbed
          itemTiles={itemTiles}
        />
      </Backdrop>
    )
  },
  {
    id: 'moved',
    label: 'Tile moved in list',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <Backdrop>
        <AppLauncherModal
          appTiles={appTilesMoved}
          dragDropLiveRegion="Sales Cloud: new position 3 of 6."
          itemTiles={itemTiles}
        />
      </Backdrop>
    )
  },
  {
    id: 'dropped',
    label: 'Tile dropped',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <Backdrop>
        <AppLauncherModal
          appTiles={appTilesDropped}
          dragDropLiveRegion="Sales Cloud: final position 4 of 6."
          itemTiles={itemTiles}
        />
      </Backdrop>
    )
  },
  {
    id: 'search',
    label: 'Search',
    demoStyles: exampleDemoStyles,
    storybookStyles: true,
    element: (
      <Backdrop>
        <AppLauncherModal
          appTiles={appTilesSearch}
          dragDropLiveRegion=""
          itemTiles={searchItemTiles}
          searchTerm="sales"
        />
      </Backdrop>
    )
  },
  {
    id: 'tooltip',
    label: 'Tooltip',
    demoStyles: 'width: 360px; height: 13rem; position: relative; padding: 0;',
    storybookStyles: true,
    element: (
      <AppLauncherTile
        figureClass="slds-icon-custom-27"
        objectInitials="SC"
        label="Sales Cloud"
        description="The primary internal Salesforce org. Used to run our online sales business..."
        hasTooltip
        tooltipText="The primary internal Salesforce org. Used to run our online sales business and manage accounts."
        index="x"
      />
    )
  }
];
