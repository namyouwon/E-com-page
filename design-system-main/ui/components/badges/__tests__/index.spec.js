/* eslint-env jest */
import React from 'react';
import { LightestBadge, InverseBadge, Badge, BadgeIcon } from '../';
import { UtilityIcon } from '../../icons/base/example';
import createHelpers from '../../../../jest.helpers';
const { matchesMarkup } = createHelpers(__dirname);

describe('Badge colors', () => {
  it('renders a default badge', () =>
    matchesMarkup(<Badge>Lovely badge</Badge>));

  it('renders an inverse badge', () =>
    matchesMarkup(<InverseBadge>Lovely badge</InverseBadge>));

  it('renders badge', () =>
    matchesMarkup(<LightestBadge>Light</LightestBadge>));
});

describe('Badge with icons', () => {
  it('renders badge with left icon', () =>
    matchesMarkup(
      <LightestBadge>
        <BadgeIcon symbol="world" />
        Icon on the left
      </LightestBadge>
    ));

  it('renders badge with right icon', () =>
    matchesMarkup(
      <LightestBadge>
        Icon on the right
        <BadgeIcon symbol="world" />
      </LightestBadge>
    ));

  it('renders badge with icon only', () =>
    matchesMarkup(
      <LightestBadge>
        <BadgeIcon
          symbol="world"
          assistiveText="assistive text goes here, do it!"
        />
      </LightestBadge>
    ));

  it('renders deprecated badge with right icon', () =>
    matchesMarkup(
      <LightestBadge>
        Icon on the right
        <UtilityIcon
          containerClassName="slds-m-left_xx-small"
          className="slds-icon_xx-small slds-icon-text-default"
          assistiveText={false}
          symbol="world"
        />
      </LightestBadge>
    ));

  it('renders deprecated badge with icon only', () =>
    matchesMarkup(
      <LightestBadge>
        <UtilityIcon
          className="slds-icon_xx-small slds-icon-text-default"
          assistiveText="assistive text goes here, do it!"
          symbol="world"
        />
      </LightestBadge>
    ));
});
