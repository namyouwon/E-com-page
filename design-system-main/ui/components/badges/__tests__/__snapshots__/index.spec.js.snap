// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge colors renders a default badge 1`] = `
<span
  className="slds-badge"
>
  Lovely badge
</span>
`;

exports[`Badge colors renders an inverse badge 1`] = `
<span
  className="slds-badge slds-badge_inverse"
>
  Lovely badge
</span>
`;

exports[`Badge colors renders badge 1`] = `
<span
  className="slds-badge slds-badge_lightest"
>
  Light
</span>
`;

exports[`Badge with icons renders badge with icon only 1`] = `
<span
  className="slds-badge slds-badge_lightest"
>
  <span
    className="slds-badge__icon slds-badge__icon_left"
  >
    <span
      className="slds-icon_container slds-icon-utility-world slds-current-color"
      title="Description of icon when needed"
    >
      <svg
        aria-hidden={true}
        className="slds-icon slds-icon_xx-small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#world"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        assistive text goes here, do it!
      </span>
    </span>
  </span>
</span>
`;

exports[`Badge with icons renders badge with left icon 1`] = `
<span
  className="slds-badge slds-badge_lightest"
>
  <span
    className="slds-badge__icon slds-badge__icon_left"
  >
    <span
      className="slds-icon_container slds-icon-utility-world slds-current-color"
      title="Description of icon when needed"
    >
      <svg
        aria-hidden={true}
        className="slds-icon slds-icon_xx-small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#world"
        />
      </svg>
    </span>
  </span>
  Icon on the left
</span>
`;

exports[`Badge with icons renders badge with right icon 1`] = `
<span
  className="slds-badge slds-badge_lightest"
>
  Icon on the right
  <span
    className="slds-badge__icon slds-badge__icon_left"
  >
    <span
      className="slds-icon_container slds-icon-utility-world slds-current-color"
      title="Description of icon when needed"
    >
      <svg
        aria-hidden={true}
        className="slds-icon slds-icon_xx-small"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#world"
        />
      </svg>
    </span>
  </span>
</span>
`;

exports[`Badge with icons renders deprecated badge with icon only 1`] = `
<span
  className="slds-badge slds-badge_lightest"
>
  <span
    className="slds-icon_container slds-icon-utility-world"
    title="Description of icon when needed"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-icon_xx-small slds-icon-text-default"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#world"
      />
    </svg>
    <span
      className="slds-assistive-text"
    >
      assistive text goes here, do it!
    </span>
  </span>
</span>
`;

exports[`Badge with icons renders deprecated badge with right icon 1`] = `
<span
  className="slds-badge slds-badge_lightest"
>
  Icon on the right
  <span
    className="slds-icon_container slds-icon-utility-world slds-m-left_xx-small"
    title="Description of icon when needed"
  >
    <svg
      aria-hidden={true}
      className="slds-icon slds-icon_xx-small slds-icon-text-default"
    >
      <use
        xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#world"
      />
    </svg>
  </span>
</span>
`;
